### DISCUSSION TITLE
throw away DSL editor/renderer for self hosting

### DISCUSSION THREAD

**Discussion by KungFur1** [structurizr/java#330] [Ideas]
https://structurizr.com/dsl
Is currently the easiest/most accessible way to develop structurizr diagrams. Are there plans to release something like that for self hosting?

**Comment by simonbrowndotje** [comment]
Structurizr Lite is the recommended option for self-hosting, and allows you to access the full set of features support by the DSL -> https://docs.structurizr.com/dsl#comparison

**Reply by KungFur1** [reply]
I am familiar with Structurizr Lite. But using lite means using Docker. Which makes it harder to introduce this tool to new people. 

We are currently storing DSL code in a GitHub repository and using structurizr Lite for modelling/rendering purposes. We also upload workspace.json to on-premises. (We are basically doing the recommended on-premises workflow)
However in some cases Docker can become a big hurdle, so having an online throw away DSL editor would solve this problem. 
https://structurizr.com/dsl - is exactly what I am looking for. But to be able to work with sensitive data we need to host it in our own environment. 
https://structurizr.com/dsl - integrated into the on-premises version would be the best solution.

**Reply by simonbrowndotje** [reply]
You can run Lite without Docker -> https://docs.structurizr.com/lite/installation#spring-boot

**Reply by KungFur1** [reply]
Hmm.. never really tried this option, I will have to check it out. But installing java and Graphviz is still a technical challenge. Having a rendering tool accessible via link would be the best solution in our case.

**Reply by simonbrowndotje** [reply]
I've added a public DSL editor to the on-premises installation, which will be in the next release.

**Reply by KungFur1** [reply]
Thank you! That's very good news for us. Looking forward to seeing it.
